### PR TITLE
Update paused + UI

### PR DESCRIPTION
**Pull request by gabriel**
@patrickxb for core stuff
@keybase/react-hackers for UI stuff

Uses lsof to check if kbfs is in use

### CODE REVIEW

File: shared/update/index.js [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
-    undefined,
-    (stateProps, dispatchProps, ownProps) => {
-      return {
-        ...stateProps,
-        ...dispatchProps,
-        ...ownProps
-      }
-    }
-)(Update)
+export default connect()(Update)
```
Comment (chrisnojima) [inline shared/update/index.js]: does this need the entire store?

File: shared/more/component-sheet.render.desktop.js [hunk 2]
```diff
@@ -52,171 +42,48 @@ export default class Render extends Component {
...
       </div>
     )
   }
 }
 
-PopupDemo_.propTypes = {
-  dispatch: React.PropTypes.any
-}
-
-const PopupDemo = connect(state => ({}), dispatch => ({dispatch}))(PopupDemo_)
```
Comment (chrisnojima) [inline shared/more/component-sheet.render.desktop.js]: do we not use this anymore?

File: shared/update/index.js (hunk 1 above)
Comment (gabriel) [inline shared/update/index.js]: Nope fixed

File: shared/more/component-sheet.render.desktop.js (hunk 2 above)
Comment (gabriel) [inline shared/more/component-sheet.render.desktop.js]: wasn't working so I nuked it

File: shared/actions/update.js [hunk 3]
```diff
@@ -80,24 +82,43 @@ function updateListenersCreator (dispatch: Dispatch, getState: () => {config: Co
...
           alwaysUpdate: alwaysAutoInstall,
           updateCommand,
           canUpdate: !updateCommand
         }
-      }: ShowUpdateAction))
+      }: ShowUpdateConfirmAction))
+    },
+
+    'keybase.1.updateUi.updateAppInUse': (payload, response) => {
+      console.log('Update (app in use) prompt')
```
Comment (chrisnojima) [inline shared/actions/update.js]: we can probably kill all these console.logs right

File: shared/actions/update.js [hunk 4]
```diff
@@ -80,24 +82,43 @@ function updateListenersCreator (dispatch: Dispatch, getState: () => {config: Co
...
+
+      dispatch(({
+        type: Constants.showUpdatePaused,
+        payload: {
+        }
+      }: ShowUpdatePausedAction))
     },
 
     'keybase.1.updateUi.updateQuit': (param, response) => {
+      console.log('Update (quit/restart) prompt')
```
Comment (chrisnojima) [inline shared/actions/update.js]: this one too

File: shared/update/index.js [hunk 5]
```diff
@@ -1,27 +1,31 @@
...
-    undefined,
-    (stateProps, dispatchProps, ownProps) => {
-      return {
-        ...stateProps,
-        ...dispatchProps,
-        ...ownProps
-      }
-    }
+  state => {
+    return {updateConfirm: state.updateConfirm, updatePaused: state.updatePaused}
```
Comment (chrisnojima) [inline shared/update/index.js]: confused about what these props are for?

File: shared/actions/update.js (hunk 3 above)
Comment (gabriel) [inline shared/actions/update.js]: actually good to have in the log

File: shared/update/index.js (hunk 5 above)
Comment (gabriel) [inline shared/update/index.js]: updateConfirm is for the update (confirm) prompt (was renamed to that).. the updatePaused is for the (app in use/paused) prompt

Comment (chrisnojima) [inline shared/update/index.js]: but they're not used by anything in here?

Comment (gabriel) [inline shared/update/index.js]: they are used by the subcomponents? it didn't work unless I have this in... its registering to listen for those state changes? it's possible I don't understand connect though, there is some magic happening thats not totally clear to me

Comment (gabriel) [inline shared/update/index.js]: ok i understand now (marco explained to me)... will remove

**Pull request merged**